class PostFail1 {
    var a:Int := 5

    method test1(a:Int) {
        if true
            self.a := a
    }
    @post(self.a > a)
    
    @class
    method main() {
        var p := PostFail1()
        try {
            p.test1(10)
        }
        catch e:AssertionError {
            Console.writeLine("success")
        }
    }
}